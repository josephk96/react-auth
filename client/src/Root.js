import React, { useContext, useEffect } from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";

import PrivateRoute from "./components/PrivateRoute";
import Navbar from "./components/layout/Navbar";
import Home from "./components/pages/Home";
import Login from "./components/pages/Login";
import Register from "./components/pages/Register";

import AuthContext from "./context/auth/authContext";

export const Root = () => {
  const authContext = useContext(AuthContext);
  const { isAuthenticated, verifyLocalStorage } = authContext;

  useEffect(() => {
    if(!isAuthenticated && localStorage.token) {
    console.log("Checking if user is already authenticated");
    verifyLocalStorage({user: localStorage.user});
    }
  });


  return (
    <Router>
      <Navbar />
      <Switch>
        <PrivateRoute exact path="/"><Home /></PrivateRoute>
        <Route exact path="/login" component={Login} />
        <Route exact path="/register" component={Register} />
      </Switch>
    </Router>
  );
}

export default Root;