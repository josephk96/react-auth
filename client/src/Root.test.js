import React from 'react';
import Root from './Root';
import { shallow } from 'enzyme';

jest.mock('react', () => {
  const ActualReact = require.requireActual('react');
  return {
    ...ActualReact,
    useContext: () => ({
      authContext: {}
    }),
  };
});

describe('Root', () => {
  it('renders without crashing', () => {
    
    const component = shallow(<Root />);
    expect(component.debug()).toMatchSnapshot();
  });
});
