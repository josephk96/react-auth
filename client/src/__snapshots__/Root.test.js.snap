// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root renders without crashing 1`] = `
"<BrowserRouter>
  <Navbar />
  <Switch>
    <PrivateRoute exact={true} path=\\"/\\">
      <Home />
    </PrivateRoute>
    <Route exact={true} path=\\"/login\\" component={[Function: Login]} />
    <Route exact={true} path=\\"/register\\" component={[Function: Register]} />
  </Switch>
</BrowserRouter>"
`;
